### PR TITLE
Added RsStatus class with most commonly used HTTP response status codes

### PR DESCRIPTION
**Pull request by olegmoz**
Added RsStatus class to represent HTTP response status code with some commonly used statuses predefined.

### CODE REVIEW

**Comment by paulodamaso** [comment]
@g4s8 please assign me

**Comment by 0crat** [comment]
This pull request [#43](https://github.com/artipie/http/issues/43) is assigned to @fanifieiev[/z](https://www.0crat.com/u/fanifieiev), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/5cd1882b-2c1d-4730-98dc-d349d8430ec9); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/http/rs/RsStatus.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+ */
+package com.artipie.http.rs;
+
+/**
+ * HTTP response status code.
+ * See <a href="https://tools.ietf.org/html/rfc2616#section-6.1.1">RFC 2616 6.1.1 Status Code and Reason Phrase</a>
+ *
+ * @since 0.4
+ */
+public final class RsStatus {
```
Comment (fanifieiev) [inline src/main/java/com/artipie/http/rs/RsStatus.java]: @olegmoz  Well, this class looks obvious but according to EO principles, this design is bad.
Take a look at https://www.yegor256.com/2015/07/06/public-static-literals.html
@g4s8 @yegor256  WDYT???

Comment (olegmoz) [reply src/main/java/com/artipie/http/rs/RsStatus.java]: @fanifieiev talked with @yegor256 and he seems to be ok with having such class. I personally do not see any way around having such constants, it's just nature of HTTP

Comment (olegmoz) [reply src/main/java/com/artipie/http/rs/RsStatus.java]: @g4s8 what do you think?

Comment (fanifieiev) [reply src/main/java/com/artipie/http/rs/RsStatus.java]: @olegmoz Has @yegor256 approved this design???? Hm, surprised.
So, if we go this way, I propose to make this class as Enumeration, it is still static but looks better.

Comment (olegmoz) [reply src/main/java/com/artipie/http/rs/RsStatus.java]: @fanifieiev changed it to enum, please review

**Comment by 0crat** [comment]
@fanifieiev[/z](https://www.0crat.com/u/fanifieiev) this job was assigned to you 6days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by olegmoz** [comment]
@fanifieiev branch was merged with master, please review

**Comment by fanifieiev** [comment]
@g4s8 Please merge

File: src/main/java/com/artipie/http/rs/RsStatus.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+    /**
+     * Service Unavailable.
+     */
+    UNAVAILABLE("503");
+
+    /**
+     * Code value.
+     */
+    private final String string;
```
Comment (g4s8) [inline src/main/java/com/artipie/http/rs/RsStatus.java]: @olegmoz what is the point of using `String` here? Status code is a number, so I'd use `int` here.

Comment (olegmoz) [reply src/main/java/com/artipie/http/rs/RsStatus.java]: @g4s8 actually I think that `String` makes more sense - we do not do arithmetics with codes. Only thing we do with them is print to response and compare if codes are identical.

Comment (fanifieiev) [reply src/main/java/com/artipie/http/rs/RsStatus.java]: @olegmoz Kirill(@g4s8 ) is absolutely right. Sorry to have missed that.
Even though there is no much difference, but according to specification [w3org](https://www.w3.org/Protocols/rfc2616/rfc2616-sec6.html), it is required that the http status code is a 3-digit integer(have a look at the paragraph `6.1.1 Status Code and Reason Phrase`)

Comment (olegmoz) [reply src/main/java/com/artipie/http/rs/RsStatus.java]: @fanifieiev that's right about spec, but I do not see how it makes you use `int` in your program? HTTP operates with byte strings, not with numbers

**Comment by 0crat** [comment]
Job `gh:artipie/http#43` is not assigned, can't get performer

**Comment by 0crat** [comment]
This job is not in scope

**Comment by fanifieiev** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/artipie/http/issues/43#issuecomment-594091149))

@fanifieiev This is what I know about this job in [CT2E6TK9B](https://www.0crat.com/p/CT2E6TK9B), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#43](https://github.com/artipie/http/issues/43) is not in scope
  * Job [footprint](https://www.0crat.com/footprint/CT2E6TK9B?q=%7Bjob%3A%27gh%3Aartipie%2Fhttp%2343%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Pull request merged**